### PR TITLE
Tesla: Full Stack Capstone Group 

### PR DESCRIPTION
**Pull request by luccamicale**
## Tesla: Full Stack Capstone Group  💯 

##https://github.com/luccamicale/Full-Stack-Capstone-Group  👍 

## Checklist before requesting a review.
- [x] We have performed a self-review of my code.
- [x] Make sure that there are no linters errors.
- [x] Make sure that used correct Girflow.
- [x] Make sure that you documented your work in a professional way.
- [x] Follow Best practices in HTML and CSS.
- [x] Follow Best practices in Ruby.
- [x] We selected a Cars 
- [x] We created a Reserve Forms
- [x] We created a My reservations
- [x] We created a Add Cars
- [x] We created a delete Cars
- [x] Follow the layout of the provided design
- [x] We look for image and information about Tesla
- [x] Set up the repository/repositories on GitHub and use Gitflow.
- [x] You should use Postgres as your database
- [x] Use Rails to create backend API.
- [x] Use React & Redux to create frontend UI.
- [x] We choose to create everyrthing in one repo
- [x] we create a Kanban Board 
- [x] We did the deplot with Render 👍 💯 🔢 🥇 

## [LIVE DEMO](https://reserve-tesla-car.onrender.com)

landing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,111 +99,95 @@ To get a local copy up and running, follow these steps.
...
-```sh
-  cd my-project
-  gem install
-```
---->
+Run bundle install
 
 ### Usage
 
 To run the project, execute the following command:
```
Comment (Meltrust) [inline README.md]: - [x] Kindly, include in your instructions the required db commands that your users will need to run your backend app: ``rails db:create, rails db:migrate, rails db:seed``. 👍

Comment (Meltrust) [inline README.md:148]: - [x] In your setup instructions, kindly include the required `npm install` and `npm run build` commands to include the JS dependencies and to build the frontend. 👍

File: app/javascript/App.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import ReservationForm from './components/ReservationForm';
+import Login from './components/Login';
+import ProductForm from './components/ProductForm';
+import { fetchHomes } from './redux/home/Home';
+import Signup from './components/Signup';
+import Products from './components/Products';
+import { fetchUsers } from './redux/registration/Registration';
+
+
+function App() {
```
Comment (Meltrust) [inline app/javascript/App.js]: - [x]  Please, kindly refactor this and all other functions in your project to ES6 arrow functions, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered non-best practice.

You can do a search in Vscode for the "function" word, and correct as needed.  Shouldn't take you more than 5-10 min so don't worry:

image

File: app/javascript/components/Detail.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import loggg from './img/loggg.png';
+import logo from './img/logo.gif';
+
+import '../styles/Detail.css';
+
+function Detail({setProduct}) {
+  const navigate = useNavigate();
+
+  const product = useSelector((state) => state.product);
+  console.log(`product in details page= ${product}`);
```
Comment (Meltrust) [inline app/javascript/components/Detail.js]: According to the [Javascript general requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md):

image

Therefore:

- [x] Kindly, remove these console.log debugging statements present in these files in order to improve the repo's cleanliness:


image

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @luccamicale,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  Nice code organization ✔️ 
-  App is working well✔️ 
-  Good readme ✔️ 
-  Nice mobile implementation ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: README.md (hunk 1 above)
Comment (luccamicale) [reply README.md]: I fixed it

Comment (luccamicale) [reply README.md:148]: I fixed it

**Review by Meltrust — APPROVED** [review]
#### Hi @luccamicale,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Readme instructions have been improved✔️ 
- All functions are now ES6✔️ 
- Debugging code has been removed✔️ 
- Awesome job ✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**